src/App.tsx

```diff
@@ -1,10 +1,43 @@
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import classNames from 'classnames';
+import React, { useEffect, useState } from 'react';
+import { Todo } from './types/Todo';
 import { UserWarning } from './UserWarning';
-
-const USER_ID = 0;
+import { USER_ID } from './utils/fetchClient';
+import { getTodos } from './api/todos';
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [completedTodos, setCompletedTodos]
+    = useState<Todo[]>([]);
+  const [activeTodos, setActiveTodos]
+    = useState<Todo[]>([]);
+  const [visibleTodos, setVisibleTodos] = useState<Todo[]>([]);
```

Comment:
try to reimplement it in such way that you have only todos and filter state. And based on this data you will always be able to filter out needed todos. Just imagine, that you hundreds of todos, then you will need to store almost three times more todos in memory for each filter type. Also in future it might happen that new filters will be introduces and you will need to add more filtered copies of initial todos

Reply:
I had this idea but I didn't know what is more heavy: filtering every render or keeping more copies...